stko/utilities/utilities.py

```diff
@@ -1571,3 +1572,133 @@ def get_acute_vector(reference, vector):
 def get_plane_normal(points):
     centroid = points.sum(axis=0) / len(points)
     return np.linalg.svd(points - centroid)[-1][2, :]
+
+
+def has_h_atom(bond):
+    """
+    Check if a bond has a H atom.
+
+    Parameters
+    ----------
+    bond : :class:`stk.Bond`
+        Bond to test if it has a H atom.
+
+    Returns
+    -------
+    :class:`bool`
+        Returns `True` if bond has H atom.
+
+    """
+    if bond.get_atom1().get_atomic_number() == 1:
+        return True
+    if bond.get_atom1().get_atomic_number() == 1:
+        return True
+
+    return False
+
+
+def has_metal_atom(bond, metal_atoms):
+    """
+    Check if a bond has a metal atom.
+
+    Parameters
+    ----------
+    bond : :class:`stk.Bond`
+        Bond to test if it has a metal atom.
+
+    metal_atoms : :class:`.list` of :class:`stk.Atom`
+        List of metal atoms.
+
+    Returns
+    -------
+    :class:`bool`
+        Returns `True` if bond has metal atom.
+
+    """
+    if bond.get_atom1() in metal_atoms:
+        return True
+    if bond.get_atom2() in metal_atoms:
+        return True
+
+    return False
+
+
+def metal_a_no():
+
+    return chain(range(21, 31), range(39, 49), range(72, 81))
```

Comment:
This can be made into a generator using `yield from` instead of return

Reply:
Not in this case because its used for look up